examples/structured_data/advanced_techniques_for_structured_data.py

```diff
@@ -0,0 +1,472 @@
+        csv_file_path,
+        batch_size=batch_size,
+        column_names=CSV_HEADER,
+        column_defaults=COLUMN_DEFAULTS,
+        label_name=TARGET_FEATURE_NAME,
+        num_epochs=num_epochs,
+        header=True,
+        shuffle=shuffle,
+    )
+
+    return dataset.cache()
+
+
+"""
+Here we configure the parameters and implement the procedure for running a training and evaluation
+experiment given a model.
+"""
+
+train_size = 493860
+learning_rate = 0.001
+dropout_rate = 0.1
+batch_size = 265
+num_epochs = 50
+
+train_steps_per_epoch = train_size // batch_size
+hidden_units = [32, 32]
+
+
+def run_experiment(model):
+
+    model.compile(
+        optimizer=keras.optimizers.Adam(learning_rate=learning_rate),
+        loss=keras.losses.SparseCategoricalCrossentropy(),
+        metrics=[keras.metrics.SparseCategoricalAccuracy()],
+    )
+
+    train_dataset = get_dataset_from_csv(train_data_file, batch_size, shuffle=True)
+
+    test_dataset = get_dataset_from_csv(test_data_file, batch_size, num_epochs=1)
+
+    print("Start training the model...")
+    history = model.fit(
+        train_dataset, epochs=num_epochs, steps_per_epoch=train_steps_per_epoch
+    )
+    print("Model training finished")
+
+    _, accuracy = model.evaluate(test_dataset, verbose=0)
+
+    print(f"Test accuracy: {round(accuracy * 100, 2)}%")
+
+
+"""
+## Create model inputs
+
+We define the inputs for our models as a dictionary, where the key is the feature name, and the value
+is a `keras.layers.Input` tensor with the corresponsing feature shape and data type.
+"""
+
+
+def create_model_inputs():
+    inputs = {}
+    for feature_name in FEATURE_NAMES:
+        if feature_name in NUMERIC_FEATURE_NAMES:
+            inputs[feature_name] = layers.Input(
+                name=feature_name, shape=(), dtype=tf.float32
+            )
+        else:
+            inputs[feature_name] = layers.Input(
+                name=feature_name, shape=(), dtype=tf.string
+            )
+    return inputs
+
+
+"""
+## Create feature columns
+
+We create two representations of our input features: sparse and dense:
+1. In the **sparse** representation, the categorical features are encoded with one-hot encoding,
+using [tf.feature_columns.indicator_column](https://www.tensorflow.org/api_docs/python/tf/feature_column/indicator_column).
+This representation can be useful for the model to *memorize* particular feature values to make certain
+predictions.
+2. In the **dense** representation, the categorical features are encoded with low-dimensional embeddings,
+using the [tf.feature_column.embedding_column](https://www.tensorflow.org/api_docs/python/tf/feature_column/embedding_column).
+This representation helps the model to *generalize* well to unseen feature combinations.
+
+These feature columns are converted to a Keras dense layer using the
+[tf.keras.layers.DenseFeatures](https://www.tensorflow.org/api_docs/python/tf/keras/layers/DenseFeatures).
```

Comment:
Suggestion:

```suggestion
[`tf.keras.layers.DenseFeatures`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/DenseFeatures).
```

(code in text :https://developers.google.com/style/code-in-text?hl=en0)